django/views/generic/detail.py

```diff
@@ -170,7 +173,11 @@ def get_template_names(self):
             # If we still haven't managed to find any template names, we should
             # re-raise the ImproperlyConfigured to alert the user.
             if not names:
-                raise
+                raise ImproperlyConfigured(
+                    "SingleObjectTemplateResponseMixin requires either a definition of "
+                    "'template_name_field', a definition of 'model' or an "
+                    "implementation of 'get_template_names()'"
```

Comment:
I think continuing to mention `template_name` (as the most common option) has value here. 

Since, `SingleObjectTemplateResponseMixin` subclasses `TemplateResponseMixin` and calls `super().get_template_names()`, it's still true to say...

```suggestion
                    "SingleObjectTemplateResponseMixin requires a definition "
                    "of 'template_name', 'template_name_field', or 'model', "
                    "or an implementation of 'get_template_names()'"
```

Reply:
That's what I get for just clicking the button 😬